projects/aca-content/folder-rules/src/rule-details/conditions/rule-simple-condition.ui-component.ts

```diff
@@ -147,6 +192,63 @@ export class RuleSimpleConditionUiComponent implements ControlValueAccessor, OnD
   }
 
   ngOnDestroy() {
-    this.formSubscription.unsubscribe();
+    this.onDestroy$.next();
+    this.onDestroy$.complete();
+  }
+
+  ngOnInit(): void {
+    this.form.valueChanges.pipe(takeUntil(this.onDestroy$)).subscribe((value: any) => {
+      this.onChange(value);
+      this.onTouch();
+    });
+
+    this.fieldControl.valueChanges.pipe(distinctUntilChanged(), takeUntil(this.onDestroy$)).subscribe((field: string) => {
+      if (field === 'category') {
+        this.autoCompleteOptionsSubscription = this.form
+          .get('parameter')
+          .valueChanges.pipe(distinctUntilChanged(), debounceTime(AUTOCOMPLETE_OPTIONS_DEBOUNCE_TIME), takeUntil(this.onDestroy$))
+          .subscribe((categoryName) => {
+            this.getCategories(categoryName);
+          });
+        this.parameterControl.setValue('');
+      } else {
+        this.autoCompleteOptionsSubscription?.unsubscribe();
+      }
+    });
+  }
+
+  private getCategories(categoryName: string) {
+    this.showLoadingSpinner = true;
+    this.categoryService
+      .searchCategories(categoryName)
+      .pipe(first())
+      .subscribe((existingCategoriesResult) => {
+        this.showLoadingSpinner = false;
+        const options: AutoCompleteOption[] = existingCategoriesResult?.list?.entries?.map((rowEntry) => {
+          const option = new AutoCompleteOption();
+          option.value = rowEntry.entry.id;
+          const path = rowEntry.entry.path.name.split('/').splice(3).join('/');
+          option.displayLabel = path ? `${path}/${rowEntry.entry.name}` : rowEntry.entry.name;
+          return option;
+        });
+        if (options.length > 0) {
+          this.autoCompleteOptions = this.sortAutoCompleteOptions(options);
+        }
+      });
+  }
+
+  private sortAutoCompleteOptions(autoCompleteOptions: AutoCompleteOption[]): AutoCompleteOption[] {
+    return autoCompleteOptions.sort((option1, option2) => option1.displayLabel.localeCompare(option2.displayLabel));
+  }
+
+  autoCompleteDisplayFunction: (id: string) => string = (optionValue) =>
```

Comment:
can be shorter: 
```ts
autoCompleteDisplayFunction = (optionValue: string): string =>

Reply:
Removed unneeded types from left side

Reply:
@swapnil-verma-gl you missed string types, please see again my comment - there are string types which are missing in your code. In that case it won't check if there are proper types